src/news_type.h

```diff
@@ -129,23 +133,29 @@ struct NewsItem {
 	uint32 ref1;                 ///< Reference 1 to some object: Used for a possible viewport, scrolling after clicking on the news, and for deleting the news when the object is deleted.
 	uint32 ref2;                 ///< Reference 2 to some object: Used for scrolling after clicking on the news, and for deleting the news when the object is deleted.
 
-	void *free_data;             ///< Data to be freed when the news item has reached its end.
+	const NewsAllocatedData *data; ///< Custom data for the news item that have to be deallocated (deleted) when the news item has reached its end.
```

Comment:
Didn't you just basically make a `std::unique_ptr` here? From a quick (but not detailed :P) look, that might help you out too, and avoids the "weird" dtor :)